metapool/sample_sheet.py

```diff
@@ -8,87 +8,97 @@
 import sample_sheet
 import pandas as pd
 from types import MappingProxyType
+from metapool.literals import parse_project_name, \
+    get_qiita_id_from_project_name, \
+    SAMPLES_DETAILS_KEY, SAMPLE_NAME_KEY, SAMPLE_PROJECT_KEY, \
+    CONTAINS_REPLICATES_KEY, ORIG_NAME_KEY, EXPT_DESIGN_DESC_KEY, \
+    PM_PROJECT_NAME_KEY, PM_PROJECT_PLATE_KEY, PM_BLANK_KEY
 from metapool.metapool import (bcl_scrub_name, sequencer_i5_index,
-                               REVCOMP_SEQUENCERS)
+                               REVCOMP_SEQUENCERS, TUBECODE_KEY)
 from metapool.plate import ErrorMessage, WarningMessage, PlateReplication
-from metapool.controls import SAMPLE_NAME_KEY, SAMPLE_CONTEXT_COLS, \
+from metapool.controls import SAMPLE_CONTEXT_COLS, \
     get_all_projects_in_context, is_blank, get_controls_details_from_context, \
+    get_delimited_controls_details_from_compressed_plate, \
     make_manual_control_details
 
-BIOINFORMATICS_KEY = 'Bioinformatics'
-CONTACT_KEY = 'Contact'
-SAMPLE_CONTEXT_KEY = 'SampleContext'
+_BIOINFORMATICS_KEY = 'Bioinformatics'
+_CONTACT_KEY = 'Contact'
+_SAMPLE_CONTEXT_KEY = 'SampleContext'
+_HEADER_KEY = 'Header'
+_READS_KEY = 'Reads'
+_SETTINGS_KEY = 'Settings'
+_DATA_KEY = 'Data'
+_ASSAY_KEY = 'Assay'
+_SS_SAMPLE_PROJECT_KEY = 'Sample_Project'
+_SS_QIITA_ID_KEY = 'QiitaID'
+_SS_SAMPLE_NAME_KEY = 'Sample_Name'
+_EXPERIMENT_NAME_KEY = 'Experiment Name'
+_EMAIL_KEY = 'Email'
+_HUMAN_FILTERING_KEY = 'HumanFiltering'
+_SHEET_TYPE_KEY = 'SheetType'
+_SHEET_VERSION_KEY = 'SheetVersion'
 
 _AMPLICON = 'TruSeq HT'
-_METAGENOMIC = 'Metagenomic'
-_METATRANSCRIPTOMIC = 'Metatranscriptomic'
 _STANDARD_METAG_SHEET_TYPE = 'standard_metag'
 _STANDARD_METAT_SHEET_TYPE = 'standard_metat'
 _DUMMY_SHEET_TYPE = 'dummy_amp'
 _ABSQUANT_SHEET_TYPE = 'abs_quant_metag'
-SHEET_TYPES = (_STANDARD_METAG_SHEET_TYPE, _ABSQUANT_SHEET_TYPE,
-               _STANDARD_METAT_SHEET_TYPE)
-
-HEADER_KEY = 'Header'
-READS_KEY = 'Reads'
-SETTINGS_KEY = 'Settings'
-DATA_KEY = 'Data'
-ASSAY_KEY = 'Assay'
-SS_SAMPLE_PROJECT_KEY = 'Sample_Project'
-SS_QIITA_ID_KEY = 'QiitaID'
-SS_SAMPLE_NAME_KEY = 'Sample_Name'
+_PLACEHOLDER_EXPT_NAME = 'RKL_experiment'
+
+# NB: These two string literals are assigned to (totally independent) public
+# constants in the qp_klp Step module, and then used throughout qp_klp.  I
+# don't know if those literals and these come from the same place (in
+# which case they should share a single definition), or whether they just
+# happen coincidentally to be the same, in which case they probably shouldn't.
+_METAGENOMIC = 'Metagenomic'
```

Comment:
This is very good to know!  One of these days (but probably not today :D ), it would be nice to get them using a single, shared definition, which shouldn't be a big modification: several modules of `qp_klp` already import the `metapool` module, so hopefully this change will just be importing a couple of additional constants :)